internal/resource/url_test.go

```diff
@@ -281,34 +289,34 @@ func TestParseARN(t *testing.T) {
 			key:    "resources/anonymous",
 		},
 		{
-			url:        "arn:aws:s3:us-west-2:123456789012:accesspoint/test/object",
+			url:        "arn:aws:s3:us-west-2:123456789012:accesspoint/test/object/name",
 			bucket:     "arn:aws:s3:us-west-2:123456789012:accesspoint/test",
-			key:        "object",
+			key:        "name",
 			region:     "us-west-2",
 			regionHint: "us-east-1",
 		},
 		{
-			url:        "arn:aws-cn:s3:cn-northwest-1:123456789012:accesspoint/test/object",
+			url:        "arn:aws-cn:s3:cn-northwest-1:123456789012:accesspoint/test/object/name",
 			bucket:     "arn:aws-cn:s3:cn-northwest-1:123456789012:accesspoint/test",
-			key:        "object",
+			key:        "name",
 			region:     "cn-northwest-1",
 			regionHint: "cn-north-1",
 		},
 		{
-			url:        "arn:aws-us-gov:s3:us-gov-east-1:123456789012:accesspoint/test/object",
+			url:        "arn:aws-us-gov:s3:us-gov-east-1:123456789012:accesspoint/test/object/name",
 			bucket:     "arn:aws-us-gov:s3:us-gov-east-1:123456789012:accesspoint/test",
-			key:        "object",
+			key:        "name",
 			region:     "us-gov-east-1",
 			regionHint: "us-gov-west-1",
 		},
 		{
-			url:    "arn:invalid:s3:us-west-2:123456789012:accesspoint/test/object",
+			url:    "arn:invalid:s3:us-west-2:123456789012:accesspoint/test/object/name",
 			bucket: "arn:invalid:s3:us-west-2:123456789012:accesspoint/test",
-			key:    "object",
+			key:    "name",
 			region: "us-west-2",
 		},
 		{
-			url:        "arn:aws:s3:us-west-2:123456789012:accesspoint/test/path/object",
+			url:        "arn:aws:s3:us-west-2:123456789012:accesspoint/test/object/path/object",
```

Comment:
`path/object` is the key here? Might be less confusing if it was `path/name`. Probably not worth any changes. Just noting it.

Reply:
It was intentional, actually, to make sure we don't have any weird parsing things if there's more than one `object` string.